### ISSUE TITLE
pkg-config returned flags we don't understand: -pthread -pthread

### ISSUE THREAD

**Issue by osynge** [PyAV-Org/PyAV#720]
## Overview
When installing PyAV with the following command:

    pip install av --no-binary av 

The log shows the error:

    pkg-config returned flags we don't understand: -pthread -pthread

This is due to incorrect passing of the result of `pkg-config` which outputs lines like:

    root@cd3ee8976f50:/# pkg-config --cflags --libs  libavformat
    -lavformat -lm -lbz2 -lz -lavcodec -pthread -lm -llzma -lcrystalhd -lz -lva -ldl -lswresample -lm -lavutil -pthread -lva-drm -lva -lva-x11 -lva -lvdpau -lX11 -lm -lva -lXv -lX11 -lXext -ldl
    root@cd3ee8976f50:/# pkg-config --cflags --libs  libavcodec
    -lavcodec -pthread -lm -llzma -lcrystalhd -lz -lva -ldl -lswresample -lm -lavutil -pthread -lva-drm -lva -lva-x11 -lva -lvdpau -lX11 -lm -lva -lXv -lX11 -lXext -ldl
    root@cd3ee8976f50:/# pkg-config --cflags --libs  libavdevice
    -lavdevice -lm -lxcb -lxcb-shm -lxcb-shape -lxcb-xfixes -lasound -lSDL2 -lsndio -lXv -lX11 -lXext -lavfilter -pthread -lm -lnppig -lnppicc -lnppc -lnppidei -lva -ldl -lswscale -lm -lavformat -lm -lbz2 -lz -lavcodec -pthread -lm -llzma -lcrystalhd -lz -lva -ldl -lswresample -lm -lavutil -pthread -lva-drm -lva -lva-x11 -lva -lvdpau -lX11 -lm -lva -lXv -lX11 -lXext -ldl
    root@cd3ee8976f50:/# pkg-config --cflags --libs  libavutil
    -lavutil -pthread -lva-drm -lva -lva-x11 -lva -lvdpau -lX11 -lm -lva -lXv -lX11 -lXext -ldl
    root@cd3ee8976f50:/# pkg-config --cflags --libs  libavfilter
    -lavfilter -pthread -lm -lnppig -lnppicc -lnppc -lnppidei -lva -ldl -lswscale -lm -lavformat -lm -lbz2 -lz -lavcodec -pthread -lm -llzma -lcrystalhd -lz -lva -ldl -lswresample -lm -lavutil -pthread -lva-drm -lva -lva-x11 -lva -lvdpau -lX11 -lm -lva -lXv -lX11 -lXext -ldl
    root@cd3ee8976f50:/# pkg-config --cflags --libs  libswscale
    -lswscale -lm -lavutil -pthread -lva-drm -lva -lva-x11 -lva -lvdpau -lX11 -lm -lva -lXv -lX11 -lXext -ldl

## Expected behavior

The build should not fail when `-pthread` is returned by `pkg-config --cflags --libs`

## Actual behavior

The build fails when `-pthread` is returned by `pkg-config --cflags --libs`

## Versions

- OS: 
```# lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 18.04.5 LTS
Release:        18.04
Codename:       bionic
```
- PyAV build:
```
    # python setup.py config --verbose
    running config
    PyAV: 8.0.2 (unknown commit)
    Python: 3.6.9 (default, Jul 17 2020, 12:50:27) \n[GCC 8.4.0]
    platform: Linux-4.15.0-76-generic-x86_64-with-Ubuntu-18.04-bionic
    extension_extra:
            include_dirs: [b'include']
            libraries: [b'avformat', b'm', b'bz2', b'z', b'avcodec', b'lzma', b'crystalhd', b'va', b'dl', b'swresample', b'avutil', b'va-drm', b'va-x11', b'vdpau', b'X11', b'Xv', b'Xext', b'avdevice', b'xcb', b'xcb-shm', b'xcb-shape', b'xcb-xfixes', b'asound', b'SDL2', b'sndio', b'avfilter', b'nppig', b'nppicc', b'nppc', b'nppidei', b'swscale']
            library_dirs: []
            threads: [b'thread']
            define_macros: []
            runtime_library_dirs: []
    config_macros:
            PYAV_COMMIT_STR="unknown-commit"
            PYAV_VERSION=8.0.2
            PYAV_VERSION_STR="8.0.2"

```
- FFmpeg:
```
 LD_LIBRARY_PATH=/usr/local/cuda-11.1/targets/x86_64-linux/lib/:$LD_LIBRARY_PATH  ffmpeg -version
ffmpeg version n4.2.2 Copyright (c) 2000-2019 the FFmpeg developers
built with gcc 7 (Ubuntu 7.5.0-3ubuntu1~18.04)
configuration: --prefix=/usr --enable-nonfree --enable-cuda-nvcc --enable-libnpp --extra-cflags='-I/usr/local/cuda/include -I/usr/local/cuda-11.1/targets/x86_64-linux/include/' --extra-ldflags='-L/usr/local/cuda/lib64 -L/usr/local/cuda-11.1/targets/x86_64-linux/lib/'
libavutil      56. 31.100 / 56. 31.100
libavcodec     58. 54.100 / 58. 54.100
libavformat    58. 29.100 / 58. 29.100
libavdevice    58.  8.100 / 58.  8.100
libavfilter     7. 57.100 /  7. 57.100
libswscale      5.  5.100 /  5.  5.100
libswresample   3.  5.100 /  3.  5.100
```

## Research

I have done the following:

- [Y] Checked the [PyAV documentation](https://pyav.org/docs)
- [Y] Searched on [Google](https://www.google.com/search?q=pyav+how+do+I+foo)
- [Y] Searched on [Stack Overflow](https://stackoverflow.com/search?q=pyav)
- [Y] Looked through [old GitHub issues](https://github.com/PyAV-Org/PyAV/issues?&q=is%3Aissue)
- [N] Asked on [PyAV Gitter](https://gitter.im/PyAV-Org)
- [N] ... and waited 72 hours for a response.

## Additional context

I am using this library on the docker image:

```tensorflow/tensorflow:2.3.1-gpu```

**Comment by juliantrue** [comment]
I ran into this same problem as well. I built ffmpeg with cuda support and turned out that I had a built it as a static library. Did some digging and found out that that is not supported #510. I rebuilt ffmpeg as a shared library with the `--enable-shared` flag shown below and installed again with the `--no-binary` flag and it worked a treat. Hope this helps!

```
./configure \
      --enable-shared \
      --enable-nonfree \
      --enable-cuda-nvcc \
      --enable-libnpp \
      --extra-cflags=-I/usr/local/cuda/include \
      --extra-ldflags=-L/usr/local/cuda/lib64 && \
      make -j 8
```
